lib/validator.js

```diff
@@ -0,0 +1,340 @@
+"use strict";
+
+var validator = {};
+
+var validTypes = ['blob', 'set<blob>',
+    'decimal', 'set<decimal>',
+    'double', 'set<double>',
+    'float', 'set<float>',
+    'boolean', 'set<boolean>',
+    'int', 'set<int>',
+    'varint', 'set<varint>',
+    'string', 'set<string>',
+    'timeuuid', 'set<timeuuid>',
+    'uuid', 'set<uuid>',
+    'timestamp', 'set<timestamp>',
+    'json', 'set<json>'
+];
+
+var validPredKeys = ['eq', 'lt', 'gt', 'le', 'ge', 'between'];
+
+function checkIfArrayIsUnique(arr) {
+    return new Set(arr).size === arr.length;
+}
+
+/**
+ * Validates index schemas array and throws {Error} if some preconditions are failed.
+ *
+ * The following preconditions are validated:
+ * - Index should be non-empty array
+ * - Index cannot have duplicate attributes
+ * - Index must have at least one 'hash' attribute
+ * - All indexed attributes must exist in a schema
+ * - Static indexes cannot be created on a table with no range indexes
+ *
+ * @param schema the original complete table schema
+ * @param index indexes array that's being validated
+ * @returns {Object} updated indexes array
+ */
+function validateIndexSchema(schema, index) {
+    if (!Array.isArray(index) || !index.length) {
+        throw new Error("Invalid index " + JSON.stringify(index));
+    }
+
+    var attrArray = index.map(function(ind) {
+        return ind.attribute;
+    });
+    if (!checkIfArrayIsUnique(attrArray)) {
+        throw new Error("Invalid index. Duplicate index entries.");
+    }
+
+    var haveHash = false;
+
+    index.forEach(function(elem) {
+        if (!schema.attributes[elem.attribute]) {
+            throw new Error('Index element ' + JSON.stringify(elem)
+                + ' is not in attributes!');
+        }
+
+        switch (elem.type) {
+            case 'hash':
+                haveHash = true;
+                break;
+            case 'range':
+                if (!elem.order) {
+                    // Default to descending sorting.
+                    //
+                    // Normally you should specify the sorting explicitly. In
+                    // particular, you probably always want to use descending
+                    // order for time series data (timeuuid) where access to the
+                    // most recent data is most common.
+                    elem.order = 'desc';
+                }
+                if (elem.order !== 'asc' && elem.order !== 'desc') {
+                    elem.order = 'desc';
```

Comment:
Throw an error instead of forcing, so that the schema user knows there's something wrong with it instead of living in blissful ignorance :)